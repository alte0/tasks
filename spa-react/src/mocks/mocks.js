export const myTasks = [
    {
        task_title: "Задача 1",
        task_desc: "Описание задачи: Lorem ipsum dolor sit amet, consectetur adipisicing elit. Rem alias qutask_idem officia quisquam optio, nam itaque, task_id neque fugit et tempora saepe, consequatur voluptatem maiores? Corrupti maiores provtask_ident blanditiis voluptas.",
        task_date_start:" 2019-12-12",
        task_date_end: "2019-12-13",
        task_status: "В работе",
        author_name: "Павел",
        author_surname: "Попов",
        author_patronymic: "Петрович",
        executor_name: "Павел 2",
        executor_surname: "Попов 2",
        executor_patronymic: "Петрович 2",
        task_date_no_limit: 0,
        task_id: '1'
    },
    {
        task_title: "Задача 2",
        task_desc: "Описание задачи: Lorem ipsum dolor sit amet, consectetur adipisicing elit. Rem alias qutask_idem officia quisquam optio, nam itaque, task_id neque fugit et tempora saepe, consequatur voluptatem maiores? Corrupti maiores provtask_ident blanditiis voluptas.",
        task_date_start:" 2019-10-10",
        task_date_end: "2019-10-11",
        task_status: "В работе",
        author_name: "Павел",
        author_surname: "Попов",
        author_patronymic: "Петрович",
        executor_name: "Павел 2",
        executor_surname: "Попов 2",
        executor_patronymic: "Петрович 2",
        task_date_no_limit: 0,
        task_id: '2'
    },
    {
        task_title: "Задача 3",
        task_desc: "Описание задачи: Lorem ipsum dolor sit amet, consectetur adipisicing elit. Rem alias qutask_idem officia quisquam optio, nam itaque, task_id neque fugit et tempora saepe, consequatur voluptatem maiores? Corrupti maiores provtask_ident blanditiis voluptas.",
        task_date_start:" 2019-12-29",
        task_date_end: "2019-12-29",
        task_status: "В работе",
        author_name: "Павел",
        author_surname: "Попов",
        author_patronymic: "Петрович",
        executor_name: "Павел 2",
        executor_surname: "Попов 2",
        executor_patronymic: "Петрович 2",
        task_date_no_limit: 0,
        task_id: '3'
    },
    {
        task_title: "Задача 4",
        task_desc: "Описание задачи: Lorem ipsum dolor sit amet, consectetur adipisicing elit. Rem alias qutask_idem officia quisquam optio, nam itaque, task_id neque fugit et tempora saepe, consequatur voluptatem maiores? Corrupti maiores provtask_ident blanditiis voluptas.",
        task_date_start:" 2019-12-29",
        task_date_end: "2019-12-30",
        task_status: "В работе",
        author_name: "Павел",
        author_surname: "Попов",
        author_patronymic: "Петрович",
        executor_name: "Павел 2",
        executor_surname: "Попов 2",
        executor_patronymic: "Петрович 2",
        task_date_no_limit: 0,
        task_id: '4'
    },
    {
        task_title: "Задача 5",
        task_desc: "Описание задачи: Lorem ipsum dolor sit amet, consectetur adipisicing elit. Rem alias qutask_idem officia quisquam optio, nam itaque, task_id neque fugit et tempora saepe, consequatur voluptatem maiores? Corrupti maiores provtask_ident blanditiis voluptas.",
        task_date_start:" 2019-12-29",
        task_date_end: "2019-12-31",
        task_status: "В работе",
        author_name: "Павел",
        author_surname: "Попов",
        author_patronymic: "Петрович",
        executor_name: "Павел 2",
        executor_surname: "Попов 2",
        executor_patronymic: "Петрович 2",
        task_date_no_limit: 0,
        task_id: '5'
    },
];

export const designatedTasks = [
    {
        task_title: "Задача 11",
        task_desc: "Описание задачи: Lorem ipsum dolor sit amet, consectetur adipisicing elit. ",
        task_date_start:" 2019-12-12",
        task_date_end: "2019-12-13",
        task_status: "В работе",
        author_name: "Павел",
        author_surname: "Попов",
        author_patronymic: "Петрович",
        executor_name: "Павел 2",
        executor_surname: "Попов 2",
        executor_patronymic: "Петрович 2",
        task_date_no_limit: 0,
        task_id: '1'
    },
    {
        task_title: "Задача 22",
        task_desc: "Описание задачи: Lorem ipsum dolor sit amet, consectetur adipisicing elit. ",
        task_date_start:" 2019-10-10",
        task_date_end: "2019-10-11",
        task_status: "В работе",
        author_name: "Павел",
        author_surname: "Попов",
        author_patronymic: "Петрович",
        executor_name: "Павел 2",
        executor_surname: "Попов 2",
        executor_patronymic: "Петрович 2",
        task_date_no_limit: 0,
        task_id: '2'
    },
    {
        task_title: "Задача 33",
        task_desc: "Описание задачи: Lorem ipsum dolor sit amet, consectetur adipisicing elit. ",
        task_date_start:" 2019-12-29",
        task_date_end: "2019-12-29",
        task_status: "В работе",
        author_name: "Павел",
        author_surname: "Попов",
        author_patronymic: "Петрович",
        executor_name: "Павел 2",
        executor_surname: "Попов 2",
        executor_patronymic: "Петрович 2",
        task_date_no_limit: 0,
        task_id: '3'
    },
    {
        task_title: "Задача 44",
        task_desc: "Описание задачи: Lorem ipsum dolor sit amet, consectetur adipisicing elit. ",
        task_date_start:" 2019-12-29",
        task_date_end: "2019-12-30",
        task_status: "В работе",
        author_name: "Павел",
        author_surname: "Попов",
        author_patronymic: "Петрович",
        executor_name: "Павел 2",
        executor_surname: "Попов 2",
        executor_patronymic: "Петрович 2",
        task_date_no_limit: 0,
        task_id: '4'
    },
    {
        task_title: "Задача 55",
        task_desc: "Описание задачи: Lorem ipsum dolor sit amet, consectetur adipisicing elit. ",
        task_date_start:" 2019-12-29",
        task_date_end: "2019-12-30",
        task_status: "В работе",
        author_name: "Павел",
        author_surname: "Попов",
        author_patronymic: "Петрович",
        executor_name: "Павел 2",
        executor_surname: "Попов 2",
        executor_patronymic: "Петрович 2",
        task_date_no_limit: 0,
        task_id: '5'
    },
    {
        task_title: "Задача 66",
        task_desc: "Описание задачи: Lorem ipsum dolor sit amet, consectetur adipisicing elit. ",
        task_date_start:" 2019-12-29",
        task_date_end: "2019-12-30",
        task_status: "В работе",
        author_name: "Павел",
        author_surname: "Попов",
        author_patronymic: "Петрович",
        executor_name: "Павел 2",
        executor_surname: "Попов 2",
        executor_patronymic: "Петрович 2",
        task_date_no_limit: 0,
        task_id: '6'
    },
    {
        task_title: "Задача 77",
        task_desc: "Описание задачи: Lorem ipsum dolor sit amet, consectetur adipisicing elit. ",
        task_date_start:" 2019-12-29",
        task_date_end: "2019-12-30",
        task_status: "В работе",
        author_name: "Павел",
        author_surname: "Попов",
        author_patronymic: "Петрович",
        executor_name: "Павел 2",
        executor_surname: "Попов 2",
        executor_patronymic: "Петрович 2",
        task_date_no_limit: 0,
        task_id: '7'
    },
];
