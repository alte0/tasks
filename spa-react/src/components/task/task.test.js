import React from 'react';
import renderer from 'react-test-renderer';
import Task from './task';
import {myTasks} from '../../mocks/mocks';

it('Task render', () => {
    const option = {
        isMore: false,
        isShowDesc: true
    };

    const tree = renderer
        .create(
                <Task
                    isMore={option.isMore}
                    isShowDesc={option.isShowDesc}
                    task={myTasks[0]}
                />
            )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
