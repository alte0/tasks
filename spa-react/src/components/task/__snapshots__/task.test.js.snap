// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task render 1`] = `
<section
  className="task task_expired task_single"
>
  <h3
    className="task__title"
  >
    Задача 1
  </h3>
  <div
    className="task__desc"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Описание задачи: Lorem ipsum dolor sit amet, consectetur adipisicing elit. Rem alias qutask_idem officia quisquam optio, nam itaque, task_id neque fugit et tempora saepe, consequatur voluptatem maiores? Corrupti maiores provtask_ident blanditiis voluptas.",
      }
    }
  />
  <footer
    className="task__footer"
  >
    <div
      className="task__date-start"
    >
      Начало задачи: 
       2019-12-12
    </div>
    <div
      className="task__date-end"
    >
      Закончить задачу до: 
      2019-12-13
    </div>
    <div
      className="task__status"
    >
      Статус: 
      В работе.
    </div>
    <div
      className="task__execute"
    >
      
    </div>
    <div
      className="task__author"
    >
      Назначил: 
      Попов
       
      Павел
       
      Петрович
    </div>
    <div
      className="task__executor"
    >
      Испонитель: 
      Попов 2
       
      Павел 2
       
      Петрович 2
    </div>
  </footer>
  
</section>
`;
